include/swift/Option/FrontendOptions.td

```diff
@@ -393,10 +393,6 @@ def solver_disable_shrink :
 def disable_constraint_solver_performance_hacks : Flag<["-"], "disable-constraint-solver-performance-hacks">,
   HelpText<"Disable all the hacks in the constraint solver">;
 
-def enable_operator_designated_types :
-  Flag<["-"], "enable-operator-designated-types">,
-  HelpText<"Enable operator designated types">;
-
```

Comment:
This isn't quite what I had in mind. You could leave this in and revert the change to the test that explicitly uses this, and then have ParseStdlib also set the flag below.